sync-core/src/test/java/com/cloudant/sync/replication/ResponseStreamReaderFilter.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2015 IBM Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+package com.cloudant.sync.replication;
+
+import com.cloudant.http.HttpConnection;
+import com.cloudant.http.HttpConnectionFilterContext;
+import com.cloudant.http.HttpConnectionResponseFilter;
+
+import java.io.IOException;
+
+/**
+ * Created by Rhys Short on 20/07/15.
+ */
+public class ResponseStreamReaderFilter implements HttpConnectionResponseFilter {
```

Comment:
Its not perfect, but it gives us the knowledge that the filter added to the replication object gets passed down. I would side on leaving it in, and modifying it to use new and better APIs as they come available